spec/unit/application/device_spec.rb

```diff
@@ -318,6 +320,41 @@
       expect { @device.main }.to exit_with 1
     end
 
+    it "should exit if apply is set, and target is not" do
+      @device.options.stubs(:[]).with(:apply).returns('file.pp')
+      Puppet.expects(:err).with(regexp_matches(/missing argument.* --target/))
+      expect { @device.main }.to exit_with 1
+    end
+
+    it "should error if target is passed and the apply path is incorrect" do
+      @device.options.stubs(:[]).with(:apply).returns('file.pp')
+      @device.options.stubs(:[]).with(:target).returns('device1')
+      device_hash = {
+        "device1" => OpenStruct.new(:name => "device1", :url => "ssh://user:pass@testhost", :provider => "cisco"),
+      }
+
+      Puppet::Util::NetworkDevice::Config.expects(:devices).returns(device_hash)
+      File.expects(:file?).returns(false)
+      Puppet.expects(:err).with(regexp_matches(/does not exist, cannot apply/))
+      expect { @device.main }.to exit_with 1
+    end
+
+    it "should run an apply" do
+      @device.options.stubs(:[]).with(:apply).returns('file.pp')
+      @device.options.stubs(:[]).with(:target).returns('device1')
+      device_hash = {
+        "device1" => OpenStruct.new(:name => "device1", :url => "ssh://user:pass@testhost", :provider => "cisco", :options => {}),
```

Comment:
@thallgren is that what you meant ?

Reply:
Yes. Adding that made my tests green.

Reply:
Hmm. Now the test fails because 'file.pp' isn't a file. Did something else change?